nemo_aligner/data/nlp/datasets.py

```diff
@@ -26,6 +26,57 @@
 from nemo.utils import logging
 
 
+class KnowledgeDistillationDataset(Dataset):
+    """The knowledge distillation dataset takes in raw tokens, labels, loss masks, and the teacher models' predictive top tokens & logits.
+    """
+
+    def __init__(
+        self, cfg, tokenizer, name, data_prefix, documents, data, seq_length, seed, drop_last=True,
+    ):
+        super().__init__()
+        self.cfg = cfg
+        self.name = name
+        self.data = data
+        self.seq_length = seq_length
+
+        self.nograd_length = 32
+
+        # Checks
+        assert np.min(documents) >= 0
+        assert np.max(documents) < len(self.data)
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        """Returns a pair of chosen/rejected pairs, their respective lengths, and labels.
+        """
+        payload = self.data[idx]
+        for key in ["tokens", "labels", "loss_mask", "topk_token_ids"]:
+            assert key in payload, f"{key} not in the data"
+            payload[key] = torch.tensor(payload[key], dtype=torch.int64)
+        for key in ["topk_logits", "log_sum_exp_logits"]:
+            assert key in payload, f"{key} not in the data"
+            payload[key] = torch.tensor(payload[key], dtype=torch.float32)
+
+        if self.cfg.data.top_k is not None:
+            payload["topk_logits"] = payload["topk_logits"][..., : self.cfg.data.top_k]
```

Comment:
Can you add an assertion here to make sure that the `topk_logits` are in descending order? Otherwise selecting the first `top_k` ones will leads to performance losses.